linkerd/app/src/env.rs

```diff
@@ -500,6 +520,51 @@ pub fn parse_config<S: Strings>(strings: &S) -> Result<super::Config, EnvError>
     })
 }
 
+fn oc_trace_labels(oc_file_path: Result<Option<String>, EnvError>) -> HashMap<String, String> {
+    let oc_labels = match oc_file_path {
+        Ok(path_option) => match path_option {
+            Some(path) => match fs::read_to_string(path.clone()) {
+                Ok(label_string) => convert_labels_to_map(label_string),
+                Err(err) => {
+                    warn!(
+                        "could not read OC trace labels file at path {}: {}",
+                        path, err
+                    );
+                    HashMap::<String, String>::new()
+                }
+            },
+            None => {
+                warn!("env variable {} not present", ENV_LABELS_FILE_PATH);
+                HashMap::<String, String>::new()
+            }
+        },
+        Err(err) => {
+            warn!(
+                "could not read env variable {}: {}",
+                ENV_LABELS_FILE_PATH, err
+            );
+            HashMap::<String, String>::new()
+        }
+    };
+
+    oc_labels
+}
+
+fn convert_labels_to_map(labels: String) -> HashMap<String, String> {
+    let map = labels
+        .lines()
+        .filter_map(|line| {
+            let mut parts = line.splitn(2, "=").collect::<Vec<&str>>();
+            parts.pop().and_then(move |val| {
+                parts.pop().map(move |key|
+                // Trim double quotes in value, present by default when attached through k8s downwardAPI
+                (key.to_string(), val.trim_matches('"').to_string()))
+            })
+        })
+        .collect::<HashMap<String, String>>();
```

Comment:
Nit: Since this function returns a `HashMap<String, String>`, the compiler should be able to infer the type for this `collect` without an explicit type annotation:
```suggestion
        .collect();
```